File: channel/rest/src/main/java/com/serious/service/channel/rest/AbstractRestChannelBuilder.java
package com.serious.service.channel.rest;
/*
 * @COPYRIGHT (C) 2016 Andreas Ernst
 *
 * All rights reserved
 */

import com.serious.service.ChannelManager;
import com.serious.service.channel.AbstractChannelBuilder;
import org.springframework.web.reactive.function.client.WebClient;

/**
 * @author Andreas Ernst
 */
public abstract class AbstractRestChannelBuilder extends AbstractChannelBuilder<RestChannel> {
    // constructor

    protected AbstractRestChannelBuilder(ChannelManager channelManager) {
        super(channelManager);
    }

    // protected

    abstract protected WebClient.Builder build(WebClient.Builder builder); // TODO
}

File: channel/rest/src/main/java/com/serious/service/channel/rest/AbstractRestChannelBuilder.kt
package com.serious.service.channel.rest
/*
* @COPYRIGHT (C) 2023 Andreas Ernst
*
* All rights reserved
*/

import com.serious.service.ChannelManager
import com.serious.service.channel.AbstractChannelBuilder
import org.springframework.web.reactive.function.client.WebClient

/**
 * abstract base class for [ChannelBuilder] based on a [WebClient]
 */
abstract class AbstractRestChannelBuilder  // constructor
protected constructor(channelManager: ChannelManager) : AbstractChannelBuilder<RestChannel>(channelManager) {
    // protected
    abstract fun build(builder: WebClient.Builder): WebClient.Builder
}
